packages/x-charts/src/PieChart/PieArcLabelPlot.tsx

```diff
@@ -30,11 +31,22 @@ function getItemLabel(
     return null;
   }
 
-  if (typeof arcLabel === 'string') {
-    return item[arcLabel]?.toString();
+  if (arcLabel === 'label') {
+    return getLabel(item.label, 'arc');
   }
 
-  return arcLabel(item);
+  if (arcLabel === 'value') {
+    return item.value?.toString();
+  }
+
+  if (arcLabel === 'formattedValue') {
+    return item.formattedValue;
+  }
+
+  return arcLabel({
+    ...item,
+    label: getLabel(item.label, 'arc'),
+  });
```

Comment:
Feels weird to have two levels of resolving function.

A solution could be to keep `label` as a string, and have a `labelFormatter: (item, context) => string`

The demo would then be ```labelFormatter: (item, context) => `${context.location}+${item.label}` ``` assuming items labels are A, B, and C.
In that case, we should deprecate the `arcLabel` and remove it in the next major version to keep only on way for formatting arc labels.

The other option is to create `tooltipLabel` and `legendLabel` properties. It's less elegant, but do the job

Reply:
Do you mean we should have a `labelFormatter` for the other series types as well? 

I think the issue with `arc` is that its `inner text` can actually represent values, not only the labels.

Reply:
On a related note, it would probably be very useful for us to unify the pie-charts data structure in relation to the other data structures. Though I'm not sure it is possible...